cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -501,4 +501,25 @@ void setWorkflowProgramStart(Id.Program programId, String programRunId, String w
    */
   WorkflowStatistics getWorkflowStatistics(Id.Workflow workflowId, long startTime,
                                            long endTime, List<Double> percentiles);
+
+  /**
+   * Returns the record that represents the run of a workflow.
+   *
+   * @param workflowId The Workflow whose run needs to be queried
+   * @param runId RunId of the workflow run
+   * @return A detailed report of the statistics
+   */
+  WorkflowDataset.WorkflowRunRecord getWorkflowRun(Id.Workflow workflowId, String runId);
+
+  /**
+   * Get a list of workflow runs that are spaced apart by time interval in both directions from the run id provided.
+   *
+   * @param workflow The workflow whose statistics need to be obtained
+   * @param runId The run id of the workflow
+   * @param count The count of the records that the user wants to compare against
+   * @param timeInterval The timeInterval with which the user wants to space out the runs
+   * @return Map of runId of Workflow to DetailedStatistics of the run
+   */
+  Map<String, WorkflowDataset.WorkflowRunRecord> retrieveSpacedRecords(Id.Workflow workflow, String runId,
```

Comment:
This Map was the way I was using sets to ensure uniqueness. A set of workflow records did not work out because creating a new object each time with the same params still counts as different objects. Hence, a map was used. Have converted the store to reflect collection though but internally a map is used.